src/ansys/fluent/core/fluent_connection.py

```diff
@@ -670,3 +666,20 @@ def _exit(
             remote_instance.delete()
 
         exit_event.set()
+
+
+class FluentConnectionError(ValueError):
+    """Custom Fluent connection errors."""
+
+    def __init__(self, error):
+        super().__init__(error)
+
+    @classmethod
+    def PORT_NOT_PROVIDED(cls):
```

Comment:
I'm unsure about the usefulness of creating new exception class methods for exception messages that are only raised in one location in the code, as seems to be the case with `PORT_NOT_PROVIDED`.

Regardless, if this is how we would like to structure things moving forward, I understand.

Reply:
@raph-luc Please allow me to explain the idea here via an example.  
`launch_fluent(...)` is called. Maybe I call it from my Python interpreter, or it's called from some other application (could be a cloud application). Fluent launches successfully but PyFluent fails to connect to Fluent. The calling application will catch exceptions raised by `launch_fluent`. It will use the exception information for observability and potentially for recovery. The exception type is the main source of information. Programmatically, that type is the best thing for deciding how to proceed. The message is useless in this respect if we are trying to do things rigorously. What are the useful exception types that can potentially help us to determine the recovery strategy? Does a FluentConnection error tell us anything more than a ValueError (or any other low-level exception)? Regarding the one location argument: if we can reuse exception types, that's good, but it's not a requirement.

Reply:
Thanks @seanpearsonuk, I understand that part, in this example `FluentConnectionError` can be caught, and that can therefore be a useful addition.

However, `FluentConnectionError.PORT_NOT_PROVIDED` is of little use in my understanding. It cannot be caught directly, it is just wrapping around the `"The port to connect to Fluent session is not provided."` message.

In this case, if `FluentConnectionError.PORT_NOT_PROVIDED` is only used in one location, then we are replacing what could be just `raise FluentConnectionError("The port to connect to Fluent session is not provided.")` in that location, by:
```
raise FluentConnectionError.PORT_NOT_PROVIDED()
```
plus the definition
```
    @classmethod
    def PROCESS_NOT_FINISHED(cls):
        return cls(
            "Fluent remote instance not supported by FluentConnection.wait_process_finished()."
        )
```
Which doesn't seem very efficient, to me at least. 

Am I misunderstanding or missing something?

Reply:
@raph-luc No, you are not misunderstanding. There are just some mistakes in the implementation which @hpohekar is working through. Thanks for notifying him about that.

Reply:
@raph-luc The suggestion given by @mkundu1 to separate out class methods as hierarchy will resolve your concern mentioned above.